corehq/form_processor/models.py

```diff
@@ -600,11 +602,19 @@ def __ne__(self, other):
 
     @classmethod
     def form_transaction(cls, case, xform):
+        return cls._from_form(case, xform, transaction_type=CaseTransaction.TYPE_FORM)
+
+    @classmethod
+    def ledger_transaction(cls, case, xform):
+        return cls._from_form(case, xform, transaction_type=CaseTransaction.TYPE_LEDGER)
+
+    @classmethod
+    def _from_form(cls, case, xform, transaction_type):
         return CaseTransaction(
             case=case,
             form_id=xform.form_id,
```

Comment:
I think there is a unique constraint on `(case_id, form_id)` so might need to drop that if we're going to have multiple transactions from the same form. Will also then need to update some other functions to make sure we get distinct form_ids e.g. get_case_xform_ids